enumerable.rb

```diff
@@ -34,74 +34,100 @@ def my_select
     new_arr
   end
 
-  def my_all?
-    return self unless block_given?
+  def my_checker(cond)
+    return false if cond.nil? || cond == false
 
-    w = 0
-    while w < length
-      return false unless yield(self[w])
+    true
+  end
 
-      w += 1
+  def my_all?(_arg = nil)
+    if block_given?
+      w = 0
+      while w < length
+        return false unless yield(self[w])
+
+        w += 1
+      end
+      return true
+    end
+    my_each do |x|
+      return false if my_checker(x) == false
     end
     true
   end
 
-  def my_any?
-    return self unless block_given?
-
-    i = 0
-    while i < length
-      return true if yield(self[i])
+  def my_any?(_arg = nil)
+    if block_given?
+      i = 0
+      while i < length
+        return true if yield(self[i])
 
-      i += 1
+        i += 1
+      end
+      return false
+    else
+      (0...length).each do |x|
+        return true if !self[x].nil? && self[x] != false
+      end
     end
     false
   end
 
-  def my_none?
-    return self unless block_given?
-
-    i = 0
-    while i < length
-      return false if yield(self[i])
+  def my_none?(_arg = nil)
+    if block_given?
+      i = 0
+      while i < length
+        return false if yield(self[i])
 
-      i += 1
+        i += 1
+      end
+      return true
+    else
+      (0...length).each do |x|
+        return false if self[x] != false && !self[x].nil?
+      end
     end
     true
   end
 
-  def my_count
-    return length unless block_given?
+  def my_count(arg = nil)
+    unless arg.nil?
+      count = 0
+      my_each { |value| count += value == arg ? 1 : 0 }
+      return count
+    end
+    if block_given?
 
-    i = 0
-    results = 0
-    while i < length
-      results += 1 if yield(self[i])
-      i += 1
+      i = 0
+      results = 0
+      while i < length
+        results += 1 if yield(self[i])
+        i += 1
+      end
+      results
+    else
+      length
     end
-    results
   end
 
   def my_map
-    return self unless block_given?
+    return to_enum(:my_map) unless block_given?
 
     i = 0
     new_arr = []
     while i < length
-      new_arr.push(self[i]) if yield(self[i])
+      new_arr.push(yield(self[i]))
       i += 1
     end
     new_arr
   end
 
-  def my_inject(results = 0)
-    return self unless block_given?
-
-    i = 0
-    while i < length
+  def my_inject(results = 0, symbol = nil)
+    return to_enum(:my_inject) unless block_given?
 
+    my_each { |value| results = results.send symbol, value } unless symbol.nil?
+    (0...length).each do |i|
       results = yield(results, self[i])
-      i += 1
     end
     results
   end
```

Comment:
### These tests are failing in `my_inject` method
- enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
- enumerables my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method